Build time very long

Hey,
I was trying to build the repo with an additional driver but the build time exceeded >12h and I needed to stop. I simply ran "make -j 8" on the master branch with the new driver added to src/devices. Does anyone have an idea why this happens? Maybe I'm missing something here.

So, to reproduce, if I run `make clean; make -j` on a clean tree, that works fine.  Then, I create a file `src/devices/foo.c` with the one line `#include "foo.h"` and re-run make, and I get an endless string of 

```
/home/scott/git/linuxcnc/src/Makefile.modinc:107: warning: ignoring old recipe for target 'install'
Makefile:80: warning: overriding recipe for target 'install'
/home/scott/git/linuxcnc/src/Makefile.modinc:107: warning: ignoring old recipe for target 'install'
Makefile:80: warning: overriding recipe for target 'install'
/home/scott/git/linuxcnc/src/Makefile.modinc:107: warning: ignoring old recipe for target 'install'
Makefile:80: warning: overriding recipe for target 'install'
/home/scott/git/linuxcnc/src/Makefile.modinc:107: warning: ignoring old recipe for target 'install'
Makefile:80: warning: overriding recipe for target 'install'
```

--- a/src/Makefile
+++ b/src/Makefile
@@ -40,13 +40,11 @@ $(lcec-conf-objs): EXTRA_CFLAGS := $(filter-out -Wframe-larger-than=%,$(EXTRA_CF
 #        Without this, the target for 'devices/lcec_generic.o' would just
 #        be 'lcec_generic.o', which breaks the makefile.
 #   -MM: generate dependencies, but ignore system .h files
-#   -MG: assume that missing header files are generated and add them without
-#        throwing an error.
 #   -MP: add a dummy rule for each .h that will keep make from throwing
 #        errors if headers are deleted without 'make clean'.
 %.d: %.c
 	@set -e; rm -f $@; \
-	$(CC) -MT $(subst .d,.o,$@) -MM -MG -MP $(EXTRA_CFLAGS) $< > $@.$$$$; \
+	$(CC) -MT $(subst .d,.o,$@) -MM -MP $(EXTRA_CFLAGS) $< > $@.$$$$; \
 	sed 's,\($*\)\.o[ :]*,\1.o $@ : ,g' < $@.$$$$ > $@; \
 	rm -f $@.$$$$
 

